airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -517,6 +518,14 @@ public Info info() {
    * implementation with OS processes and resources, which are automatically reaped by the OS.
    */
   private void close() {
+    // short-circuit if close was already called, so we don't re-offer ports multiple times
```

Comment:
should this be a `synchronised` method?

Reply:
or the boolean by an atomic boolean?

Reply:
I think atomic is preferred as it's hardware level while synchronise is a JVM wide locking mechanims.